File: springboot-vue-version/backend/src/main/java/com/mission5/notes/entity/Prompt.java
package com.mission5.notes.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import java.time.LocalDateTime;
import java.util.*;

@Entity
@Table(name = "prompts")
public class Prompt {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "提示词名称不能为空")
    @Size(max = 100, message = "提示词名称不能超过100个字符")
    @Column(nullable = false, length = 100)
    private String title;

    @NotBlank(message = "提示词内容不能为空")
    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @NotBlank(message = "反思备注不能为空")
    @Column(nullable = false, columnDefinition = "TEXT")
    private String notes;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(name = "prompt_tags",
        joinColumns = @JoinColumn(name = "prompt_id"),
        inverseJoinColumns = @JoinColumn(name = "tag_id"))
    private Set<Tag> tags = new HashSet<>();

    @OneToMany(mappedBy = "prompt", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    @OrderBy("createdAt DESC")
    private List<Run> runs = new ArrayList<>();

    @Size(max = 50)
    @Column(length = 50)
    private String version;

    @Min(value = 1, message = "评分最低为1")
    @Max(value = 5, message = "评分最高为5")
    private Integer rating;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    // Getters & Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title != null ? title.trim() : null; }
    public String getContent() { return content; }
    public void setContent(String content) { this.content = content != null ? content.trim() : null; }
    public String getNotes() { return notes; }
    public void setNotes(String notes) { this.notes = notes != null ? notes.trim() : null; }
    public Set<Tag> getTags() { return tags; }
    public void setTags(Set<Tag> tags) { this.tags = tags; }
    public List<Run> getRuns() { return runs; }
    public void setRuns(List<Run> runs) { this.runs = runs; }
    public String getVersion() { return version; }
    public void setVersion(String version) { this.version = version; }
    public Integer getRating() { return rating; }
    public void setRating(Integer rating) { this.rating = rating; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public LocalDateTime getUpdatedAt() { return updatedAt; }
}

File: mern-version/models/Prompt.js
const mongoose = require('mongoose');

const promptSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: [true, '提示词名称不能为空'],
      trim: true,
      maxlength: [100, '提示词名称不能超过100个字符'],
      validate: {
        validator: function (v) {
          return v.trim().length > 0;
        },
        message: '提示词名称不能全为空格',
      },
    },
    content: {
      type: String,
      required: [true, '提示词内容不能为空'],
      trim: true,
      validate: {
        validator: function (v) {
          return v.trim().length > 0;
        },
        message: '提示词内容不能全为空格',
      },
    },
    notes: {
      type: String,
      required: [true, '反思备注不能为空'],
      trim: true,
      validate: {
        validator: function (v) {
          return v.trim().length > 0;
        },
        message: '反思备注不能全为空格',
      },
    },
    tags: [
      {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Tag',
      },
    ],
    // 运行记录（可叠加多次，每次记录模型/token/耗时）
    runs: [
      {
        model: {
          type: String,
          trim: true,
          required: [true, '模型名称不能为空'],
        },
        tokens: {
          type: Number,
          min: [0, 'token 消耗不能为负数'],
          default: 0,
        },
        responseTime: {
          type: Number,
          min: [0, '响应耗时不能为负数'],
          default: 0,
        },
        createdAt: {
          type: Date,
          default: Date.now,
        },
      },
    ],
    version: {
      type: String,
      trim: true,
    },
    rating: {
      type: Number,
      min: [1, '评分最低为1'],
      max: [5, '评分最高为5'],
    },
  },
  {
    timestamps: true,
  }
);

module.exports = mongoose.model('Prompt', promptSchema);
